src/Ydb.Sdk/src/Auth/StaticCredentialsProvider.cs

```diff
@@ -0,0 +1,209 @@
+using System.IdentityModel.Tokens.Jwt;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Ydb.Sdk.Services.Auth;
+
+namespace Ydb.Sdk.Auth;
+
+public class StaticCredentialsProvider : ICredentialsProvider, IUseDriverConfig
+{
+    private readonly ILogger _logger;
+
+    private readonly string _user;
+    private readonly string _password;
+
+    private Driver? _driver;
+
+    public int MaxRetries = 5;
+
+    private readonly object _lock = new();
+
+    private volatile TokenData? _token;
+    private volatile Task? _refreshTask;
+
+    public float RefreshInterval = .5f;
+
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name="user">User of the database</param>
+    /// <param name="password">Password of the user. If user has no password use null or "" </param>
+    /// <param name="loggerFactory"></param>
+    public StaticCredentialsProvider(string user, string? password, ILoggerFactory? loggerFactory = null)
```

Comment:
No need special null case.

Null almost always - special case for need additional code for support. Is not usual value of variable type and it good place for mistakes.

Use null only if it is really need.